PdfJsViewerPlugin.php

```diff
@@ -104,6 +104,10 @@ public function submissionCallback($hookName, $args)
                 }
             }
             $templateMgr = TemplateManager::getManager($request);
+
+            if ($galleyPublication) {
+                $title = $galleyPublication->getLocalizedTitle(null, 'html');
+            }
```

Comment:
@jonasraoni Thanks for the feedback. I thought that keeping the title fetching in `PHPJsViewer.php` would maintain a clear separation of concerns, ensuring that data handling is completed before reaching the presentation layer.That way it helps keep our templates focused on presentation and avoids potential issues related to processing logic in the template.

Reply:
I also prefer to transform the data and feed primitives to templates, my suggestion was mostly to follow/keep the standard. If you decide to follow this path, then it makes sense to also set a `$title` variable on the other template.

---

Brief discussion... I think it's better to have all logic in a single place, than spread through templates, backend and database (stored procedures/triggers). Also, code analyzers will not be able to detect anything wrong inside the Smarty templates.

Notice that plugins can extend the templates through hooks (and of course, use the `object` variables). But as there were discussions regarding retiring Smarty, the topic is relevant.

Tagging @asmecher for an opinion regarding standardization.

Reply:
As a general philosophy I think the templates should have access to the object model, and be in charge of escaping/formatting data retrieved from there as they want. So for example:

Give the whole object to the Smarty template manager in PHP:
```php
$templateMgr->assign([
    'submission' => $submission,
    'issue' => $issue,
    ...
]);
```
...then let Smarty decide what to present and how to escape it:
```
<div class="title">{$submission->getLocalizedTitle(null, 'html')|strip_unsafe_html}</div>
```

This gives the most flexibility to the template in deciding what's important to present and how to properly format it. Think e.g. of themes, where templates are overridden. If assumptions are coded into the PHP handler about how the data will be presented, then themes that don't agree with those assumptions will have to work around those assumptions.

This case is a little bit weird because we're extending the `display.tpl` template with two use cases that have competing needs. I think cleaning that up will make this much easier. All that `templates/submissionGalley.tpl` and `templates/issueGalley.tpl` do is set up variables and call `display.tpl`; that can be done much better in PHP-side than  in Smarty. I'd suggest removing those two templates and setting up all the variables in PHP.

You'll have to consider how this affects the [Hypothesis plugin](https://github.com/asmecher/hypothesis) too, though -- it looks for the existing template filenames and intervenes when it sees them. So I'd suggest moving this issue to 3.5 only (since it'll break backwards compatibility) as well.

Reply:
Okay! I'm not a big fan of coding in templates, but our use-case with plugins makes it interesting and it's also more performatic and simpler than mapping data into primitives.